src/mds/CInode.cc

```diff
@@ -1546,28 +1546,39 @@ void InodeStoreBase::decode_old_inodes(bufferlist::const_iterator &p) {
 void InodeStoreBase::encode_bare(bufferlist &bl, uint64_t features,
 				 const bufferlist *snap_blob) const
 {
+  // dirfragtree
+  bufferlist dirfragtree_bl;
```

Comment:
One more comment: this secondary bufferlist will add overhead to encoding which MDS spends an inordinate amount of time doing already. Can you think of a way to pass these structures down the `encode_inode_base_bare` so that we don't need to do this.